modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/exp/ConverterUtils.java

```diff
@@ -175,6 +175,54 @@ public static Expression convert(Expression operand, Type toType) {
         return convert(operand, fromType, toType);
     }
 
+    /**
+     * Convert {@code operand} from {@code targetType} to BigDecimal type.
+     *
+     * @param operand The expression to convert
+     * @param targetType Target type
+     * @param fromType Field type
+     * @return An expression with BidDecimal type, which calls IgniteSqlFunctions.toBigDecimal function.
+     */
+    public static Expression convertToDecimal(Expression operand, Type fromType, RelDataType targetType) {
+        final Primitive fromBox = Primitive.ofBox(fromType);
+        final Primitive fromPrimitive = Primitive.of(fromType);
+        if (fromBox != null) {
+            // E.g. from "Integer" to "BigDecimal".
+            // Generate "x == null ? null : new BigDecimal(x.intValue())"
+
+            return Expressions.condition(
+                Expressions.equal(operand, RexImpTable.NULL_EXPR),
+                RexImpTable.NULL_EXPR,
+                Expressions.call(
+                    IgniteSqlFunctions.class,
+                    "toBigDecimal",
+                    Expressions.unbox(operand, fromBox),
+                    Expressions.constant(targetType.getPrecision()),
+                    Expressions.constant(targetType.getScale())));
+        }
+        if (fromPrimitive != null) {
+            // E.g. from "int" to "BigDecimal".
```

Comment:
looks like only few primitives are supported: flood and double. How is would work with other types like int or short? 

Let's write a simple test that verifies a whole spectrum of conversions to decimal types: from byte, short, int, long, float, double. For both Nullable and Non Nullable cases.

Negative cases could be covered too, like attempt to convert from bool, for example

Reply:
Added IgniteSqlFunctionsTest that covers all the cases.